mvn -e clean package failing

```
$ mvn -e clean package
[INFO] Error stacktraces are turned on.
[INFO] Scanning for projects...
[INFO]
[INFO] -------< com.theappsolutions.nanostream:NanostreamDataflowMain >--------
[INFO] Building NanostreamDataflowMain 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- maven-clean-plugin:3.0.0:clean (default-clean) @ NanostreamDataflowMain ---
[INFO] Deleting /Users/allenday/src/nanostream-dataflow/NanostreamDataflowMain/target
[INFO]
[INFO] --- maven-resources-plugin:3.0.2:resources (default-resources) @ NanostreamDataflowMain ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /Users/allenday/src/nanostream-dataflow/NanostreamDataflowMain/src/main/resources
[INFO]
[INFO] --- maven-compiler-plugin:3.7.0:compile (default-compile) @ NanostreamDataflowMain ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 45 source files to /Users/allenday/src/nanostream-dataflow/NanostreamDataflowMain/target/classes
[INFO]
[INFO] --- maven-resources-plugin:3.0.2:testResources (default-testResources) @ NanostreamDataflowMain ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 9 resources
[INFO]
[INFO] --- maven-compiler-plugin:3.7.0:testCompile (default-testCompile) @ NanostreamDataflowMain ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 13 source files to /Users/allenday/src/nanostream-dataflow/NanostreamDataflowMain/target/test-classes
[INFO]
[INFO] --- maven-surefire-plugin:2.20.1:test (default-test) @ NanostreamDataflowMain ---
[INFO] Surefire report directory: /Users/allenday/src/nanostream-dataflow/NanostreamDataflowMain/target/surefire-reports
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  5.093 s
[INFO] Finished at: 2019-02-11T01:27:50+08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.20.1:test (default-test) on project NanostreamDataflowMain: Execution default-test of goal org.apache.maven.plugins:maven-surefire-plugin:2.20.1:test failed.: NullPointerException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.20.1:test (default-test) on project NanostreamDataflowMain: Execution default-test of goal org.apache.maven.plugins:maven-surefire-plugin:2.20.1:test failed.
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:564)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-test of goal org.apache.maven.plugins:maven-surefire-plugin:2.20.1:test failed.
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:148)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:564)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: java.lang.NullPointerException
    at org.apache.maven.surefire.shade.org.apache.commons.lang3.SystemUtils.isJavaVersionAtLeast (SystemUtils.java:1626)
    at org.apache.maven.plugin.surefire.AbstractSurefireMojo.getEffectiveJvm (AbstractSurefireMojo.java:2107)
    at org.apache.maven.plugin.surefire.AbstractSurefireMojo.getForkConfiguration (AbstractSurefireMojo.java:1976)
    at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider (AbstractSurefireMojo.java:1111)
    at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked (AbstractSurefireMojo.java:954)
    at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute (AbstractSurefireMojo.java:832)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:564)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
[ERROR]
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
```

--- a/NanostreamDataflowMain/pom.xml
+++ b/NanostreamDataflowMain/pom.xml
@@ -135,6 +135,10 @@
             <plugin>
                 <artifactId>maven-surefire-plugin</artifactId>
                 <version>2.20.1</version>
+                <!-- Fixes build for OpenJDK, see: https://stackoverflow.com/questions/53010200/maven-surefire-could-not-find-forkedbooter-class -->
+                <configuration>
+                    <useSystemClassLoader>false</useSystemClassLoader>
+                </configuration>
             </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
@@ -168,14 +172,6 @@
                     </execution>
                 </executions>
             </plugin>
-            <!--<plugin>-->
-            <!--<artifactId>maven-install-plugin</artifactId>-->
-            <!--<version>2.5.2</version>-->
-            <!--</plugin>-->
-            <!--<plugin>-->
-            <!--<artifactId>maven-deploy-plugin</artifactId>-->
-            <!--<version>2.8.2</version>-->
-            <!--</plugin>-->
         </plugins>
     </build>
 </project>

--- a/NanostreamDataflowMain/src/test/java/com/theappsolutions/nanostream/output/SequenceStatisticResultTest.java
+++ b/NanostreamDataflowMain/src/test/java/com/theappsolutions/nanostream/output/SequenceStatisticResultTest.java
@@ -1,7 +1,8 @@
 package com.theappsolutions.nanostream.output;
 
 import japsa.seq.Sequence;
-import javafx.util.Pair;
+// TODO: get rid of javafx.util in order to avoid issues with OpenJDK
+//import javafx.util.Pair;
 import org.apache.beam.sdk.values.KV;
 import org.junit.Test;
 
@@ -20,19 +21,19 @@ public class SequenceStatisticResultTest {
         int sequenceMock1Quantity = 2;
         int sequenceMock2Quantity = 4;
 
-        Pair<String, String> sequenceMock1TestData = new Pair<>("sequenceMock1Name", "sequenceMock1TestValue");
-        Pair<String, String> sequenceMock2TestData = new Pair<>("sequenceMock2Name", "sequenceMock2TestValue");
+//        Pair<String, String> sequenceMock1TestData = new Pair<>("sequenceMock1Name", "sequenceMock1TestValue");
+//        Pair<String, String> sequenceMock2TestData = new Pair<>("sequenceMock2Name", "sequenceMock2TestValue");
+//
+//        Sequence sequenceMock1 = mock(Sequence.class);
+//        when(sequenceMock1.toString()).thenReturn(sequenceMock1TestData.getKey());
+//
+//        Sequence sequenceMock2 = mock(Sequence.class);
+//        when(sequenceMock2.toString()).thenReturn(sequenceMock2TestData.getKey());
 
-        Sequence sequenceMock1 = mock(Sequence.class);
-        when(sequenceMock1.toString()).thenReturn(sequenceMock1TestData.getKey());
-
-        Sequence sequenceMock2 = mock(Sequence.class);
-        when(sequenceMock2.toString()).thenReturn(sequenceMock2TestData.getKey());
-
-        List<KV<String, Sequence>> srcCollection = new ArrayList<>();
-        IntStream.range(0, sequenceMock1Quantity).forEach(index -> srcCollection.add(KV.of(sequenceMock1TestData.getKey(), sequenceMock1)));
-        IntStream.range(0, sequenceMock2Quantity).forEach(index -> srcCollection.add(KV.of(sequenceMock2TestData.getKey(), sequenceMock2)));
-        Collections.shuffle(srcCollection);
+//        List<KV<String, Sequence>> srcCollection = new ArrayList<>();
+//        IntStream.range(0, sequenceMock1Quantity).forEach(index -> srcCollection.add(KV.of(sequenceMock1TestData.getKey(), sequenceMock1)));
+//        IntStream.range(0, sequenceMock2Quantity).forEach(index -> srcCollection.add(KV.of(sequenceMock2TestData.getKey(), sequenceMock2)));
+//        Collections.shuffle(srcCollection);
 
         //TODO finish
        /* Date approximateDate = new Date();

